dts: arm: ti: Move am64x_{main,mcu} to dts/vendor/ti

dnltz opened the pull request:
This files can be used by the 32-bit as well as 64-bit ARM architectures. Move them into the dts/vendor/ti directory to make the ISA independant.

dnltz committed:
dts: arm: ti: Move am64x_{main,mcu} to dts/vendor/ti

This files can be used by the 32-bit as well as 64-bit ARM
architectures. Move them into the dts/vendor/ti directory to
make the ISA independant.

All nodes located in the AM64x MAIN domain should have the main_
prefix. This makes it more clear where those pins are actually
located in the chip.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz commented:
We currently don't have a AM64x A53x board. This is a minor clean-up before updating the AM62x device-trees to the latest AM64x changes.

dnltz commented:
@gramsay0 @glneo @vaishnavachath do we want to call all nodes in the MAIN domain `main_xxx`, similar to Linux? That's currently not implemented for the AM64x.

glneo commented:
@dnltz , yes that seems like a good idea. For any node that has instances in both MAIN and MCU domains, we should label them with the domain prefix. Having this match Linux is also a valid reason.

dnltz committed:
boards: ti: am64/am24x: Update AM64x Nodes in MAIN

Update all nodes located in the MAIN domain and add the
main_ prefix. This is required because the am64x_main.dtsi
now has this prefix.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

glneo commented:
If this goes in first I'll go update my series here to match[0], otherwise you'll need to update the timer node names in `sk_am64_am6442_r5fx_x.dts`. Should be trivial either way.

[0] https://github.com/zephyrproject-rtos/zephyr/pull/88123

glneo reviewed (commented):
You should merge patch 2 and 3 here, or else we will have a bisect build break in-between the two.

dnltz commented:
Rebased to resolve conflicts.

dnltz commented:
@vaishnavachath @gramsay0 took me a while to get back but I updated this PR and it now passes all tests.

dnltz commented:
Rebased to solve conflicts with new SPI node.

dnltz commented:
@vaishnavachath ping :)
